pkg/operator/mon/mon.go

```diff
@@ -328,18 +329,22 @@ func getNodeInfoFromNode(n v1.Node) (*NodeInfo, error) {
 			break
 		}
 	}
+	nr.Hostname = n.Labels["kubernetes.io/hostname"]
 	if nr.Address == "" {
-		return nil, fmt.Errorf("no IP given for node %s", nr.Name)
+		return nil, fmt.Errorf("no IP given for node %s", nr.Address)
```

Comment:
inside this block, `nr.Address` will be an empty string, so using that in the log message won't be as helpful as using a different (non empty) field on the `NodeInfo` struct.